src/cyclonedds/clayer/src/pysertype.c

```diff
@@ -1505,18 +1517,38 @@ ddspy_take_endpoint(PyObject *self, PyObject *args)
 
     for(int i = 0; i < (sts > N ? N : sts); ++i) {
         PyObject* sampleinfo = get_sampleinfo_pyobject(&info[i]);
+        if (PyErr_Occurred()) { 
+            PyErr_Clear();
+            PyErr_SetString(PyExc_Exception, "Sampleinfo errored.");
+            return NULL;
+        }
```

Comment:
There's quite a lot of repetition here. I'm fine with it for now, but at some point it might be worth going about this in a more practical fashion. At least, seems like this sort of thing happens all over the code.

Reply:
I agree the error handling like this is suboptimal, we should definitely pursue a better style for this, a thing for the todo list.